core/types/rollup_cost.go

```diff
@@ -259,11 +299,26 @@ func intToScaledFloat(scalar *big.Int) *big.Float {
 
 // extractL1GasParams extracts the gas parameters necessary to compute gas costs from L1 block info
 func extractL1GasParams(config *params.ChainConfig, time uint64, data []byte) (gasParams, error) {
-	// edge case: for the very first Ecotone block we still need to use the Bedrock
-	// function. We detect this edge case by seeing if the function selector is the old one
-	// If so, fall through to the pre-ecotone format
-	// Both Ecotone and Fjord use the same function selector
-	if config.IsEcotone(time) && len(data) >= 4 && !bytes.Equal(data[0:4], BedrockL1AttributesSelector) {
+	if config.IsIsthmus(time) && len(data) >= 4 && !bytes.Equal(data[0:4], EcotoneL1AttributesSelector) {
```

Comment:
Correct me if I'm wrong, but shouldn't the default logic calling `extractL1GasParamsPreEcotone()` on line 345 ([link](https://github.com/ethereum-optimism/op-geth/blob/27800ea6aac94e03410f48e3c61a9323f093170a/core/types/rollup_cost.go#L345)) be updated to target post-ecotone L1 gas param extraction?

Reply:
I don't think so, I neither Isthsmus not Ecotone is active we fallback to pre Ecotone

Reply:
I agree with @leruaa, this seems correct. Do we have a test case covering this transition? If no, we need to add it.

Did we also call out in the specs that Isthmus cannot activate on the same block as Fjord? There's an edge case where if Ecotone, Fjord and Isthmus activate at the same block, this may not work, because then the selector would be bedrock and we'd enter the first branch. We may already have a general disclaimer in the specs that Forks in general cannot activate at the same post-Genesis block.